### PR TITLE
Fix incompatible (c.f. int) initialization of mpz from strings

### LINKED ISSUE #381: Incompatible (with int) parsing of the sign in the mpz constructor
```
>>> int('+1')
1
>>> mpz('+1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid digits
```
while
```
>>>>> int('-1')
-1
>>> mpz('-1')
mpz(-1)
```
"Optionally, the string can be preceded by + or - (with no space in between), have leading zeros, be surrounded by whitespace, and have single underscores interspersed between digits." (c) CPython docs

### PR DESCRIPTION
**Pull request by skirpichev**
Closes #381

### CODE REVIEW

File: src/gmpy2_convert.c [hunk 1]
```diff
@@ -166,45 +166,61 @@ GMPy_RemoveUnderscoreASCII(PyObject *s)
...
+    PyObject *ascii_str;
+
+    ascii_str = GMPy_RemoveUnderscoreASCII(s);
 
     if (!ascii_str) return -1;
 
-    len = PyBytes_Size(ascii_str);
     cp = PyBytes_AsString(ascii_str);
 
+    while (*cp == ' ') cp++;
```
Comment (skirpichev) [inline src/gmpy2_convert.c:180]: I wonder if we shouldn't modify GMPy_RemoveUnderscoreASCII() helper to strip white spaces too.

Comment (casevh) [reply src/gmpy2_convert.c:180]: I think removing underscore and spaces in one pass makes sense.

Comment (skirpichev) [reply src/gmpy2_convert.c:180]: I had some doubts about the mpc case...

Ok, I'll do this.  (Not sure how to rename GMPy_RemoveUnderscoreASCII() helper...)

**Comment by skirpichev** [comment]
White space handling was moved to the GMPy_RemoveIgnoredASCII() helper (former GMPy_RemoveUnderscoreASCII()).

**Comment by casevh** [comment]
Thanks. Sorry about the delays in merging.

Case

**Comment by skirpichev** [comment]
> Sorry about the delays in merging.

@casevh, no need for apologize.  It's your time and thanks for investing it to the gmpy2 project.  Personally, I would prefer careful review to a quick merge...
(Sorry for a late answer.)

**Pull request merged**